lib/testutils/src/lib.rs

```diff
@@ -340,6 +339,88 @@ impl TestWorkspace {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub enum DirEntry {
+    File { contents: Vec<u8>, executable: bool },
+    Symlink { target: String },
+    GitSubmodule { commit: CommitId },
+}
+
+impl DirEntry {
+    pub fn empty_file() -> Self {
+        DirEntry::File {
+            contents: vec![],
+            executable: false,
+        }
+    }
+
+    pub fn text_file(contents: &str) -> Self {
+        DirEntry::File {
+            contents: contents.as_bytes().to_vec(),
+            executable: false,
+        }
+    }
+
+    pub fn binary_file(contents: Vec<u8>) -> Self {
+        DirEntry::File {
+            contents,
+            executable: false,
+        }
+    }
+
+    pub fn symlink(target: &str) -> Self {
+        DirEntry::Symlink {
+            target: target.to_owned(),
+        }
+    }
+
+    pub fn submodule(commit: CommitId) -> Self {
+        DirEntry::GitSubmodule { commit }
+    }
+
+    pub fn executable(self, executable: bool) -> Self {
+        match self {
+            Self::File { contents, .. } => Self::File {
+                contents,
+                executable,
+            },
+            DirEntry::Symlink { .. } => panic!("symlinks cannot be executable"),
+            DirEntry::GitSubmodule { .. } => panic!("submodules cannot be executable"),
+        }
+    }
+
+    pub fn is_executable(&self) -> bool {
+        match self {
+            DirEntry::File { executable, .. } => *executable,
+            DirEntry::Symlink { .. } => panic!("symlinks cannot be executable"),
+            DirEntry::GitSubmodule { .. } => panic!("submodules cannot be executable"),
+        }
+    }
```

Comment:
nit: This doesn't seem used, so remove it and maybe it later if we need it? It would be nice if we don't even need this function with the panics. Maybe any callers can use `match` instead?

Reply:
`is_executable()` and `text_contents()` were indeed unused. I added them when originally sketching out the API for `DirEntry`. I agree it's better to only add them later when the need arises and that `match` is the better option when it comes to the `executable` bit.